components/contentblocks/indicator-chart/DashboardIndicatorLineChartBlock.tsx

```diff
@@ -0,0 +1,263 @@
+              [String(raw[0][0]), raw[0][1]],
+              [String(raw[0][0] + 1), raw[0][1]],
+            ]
+          : raw.map(([y, v]) => [String(y), v] as [string, number]);
+
+      return {
+        name,
+        type: 'line' as const,
+        data,
+        showLine: true,
+        showSymbol: true,
+        symbolSize: 8,
+        smooth: raw.length > 1,
+        lineStyle: { width, color },
+        itemStyle: { color },
+      };
+    });
+  }
+
+  const seriesLines = buildLines(dimSeries);
+  const seriesTotal =
+    showTotalLine && totalRaw.length ? buildLines([totalDef], 3) : [];
+
+  // Trend line
+  const regData = totalRaw.slice(-Math.min(totalRaw.length, 10));
+  const predictedYears = regData.map(([yr]) => yr);
+  if (indicator?.goals?.length) {
+    const highestGoalYear = Math.max(
+      ...indicator.goals.map((g) => new Date(g.date).getFullYear())
+    );
+    if (highestGoalYear > predictedYears[predictedYears.length - 1]) {
+      predictedYears.push(highestGoalYear);
+    }
+  }
+  const model = linearRegression(regData);
+  const predictedValues = predictedYears.map((yr) => model.m * yr + model.b);
+
+  const trendSeries =
+    regData.length >= 2 &&
+    (graphsTheme.showTrendline ?? indicator?.showTrendline)
+      ? [
+          {
+            name: 'Trend',
+            type: 'line' as const,
+            symbol: 'none',
+            showSymbol: false,
+            smooth: false,
+            data: predictedYears.map(
+              (yr, i) => [String(yr), predictedValues[i]] as [string, number]
+            ),
+            lineStyle: {
+              type: 'dashed',
+              width: 2,
+              color: graphsTheme.trendLineColor ?? '#aaa',
+            },
+            itemStyle: {
+              color: graphsTheme.trendLineColor ?? '#aaa',
+            },
+            tooltip: { show: false },
+          },
+        ]
+      : [];
+
+  const goalSeries =
+    indicator?.goals?.map((g) => {
+      const y = new Date(g.date).getFullYear();
+      return {
+        name: 'Goal',
+        type: 'scatter' as const,
+        symbol: X_SYMBOL,
+        symbolSize: 10,
+        data: [[String(y), g.value]],
+        itemStyle: { color: graphsTheme.goalLineColors?.[0] ?? '#3E9C88' },
+        tooltip: { formatter: `Goal: ${g.value} ${unit}` },
+      };
+    }) || [];
+
+  const legendData = [
+    ...dimSeries.map((d) => d.name),
+    ...(seriesTotal.length ? ['Total'] : []),
+    ...(goalSeries.length ? ['Goal'] : []),
+    ...(trendSeries.length ? ['Trend'] : []),
+  ];
+
+  const option: ECOption = {
+    legend: {
+      show: true,
+      data: legendData,
+      left: 'center',
+      bottom: 0,
+      textStyle: { color: theme.textColor.secondary },
+    },
+    tooltip: {
+      trigger: 'axis',
+      appendTo: 'body',
+      axisPointer: { type: 'line' },
+      formatter: (params) => {
+        const seen = new Set<string>();
+        const ary = Array.isArray(params) ? params : [params];
```

Comment:
Nitpick: I found this a bit hard to read at first because of the naming, suggestion: 
```suggestion
        const processedSeries = new Set<string>();
        const paramsArray = Array.isArray(params) ? params : [params];
```